src/types.ts

```diff
@@ -3199,6 +3217,67 @@ export interface FollowComp extends Comp {
 export interface MoveComp extends Comp {
 }
 
+export interface OutOfViewOpt {
+	/**
+	 * If hide object when out of view.
+	 */
+	hide?: boolean,
+	/**
+	 * If pause object when out of view.
+	 */
+	pause?: boolean,
+	/**
+	 * If destroy object when out of view.
+	 */
+	destroy?: boolean,
+	/**
+	 * The screen bound offset.
+	 */
+	offset?: Vec2,
```

Comment:
Not sure about this name, maybe call it `margin`?

Reply:
nop, offset is better (my oppinion)